### ISSUE TITLE
Reverse Proxy to allow Regular Miners to mine for Coin Hive

### ISSUE THREAD

**Issue by billygerhard** [cazala/coin-hive-stratum#140]
Is it possible to use the proxy backwards, or maybe it would be a new project itself, but be able to use any miner like xmr-stak, but hash towards your coin-hive account instead of using the javascript miner for coin-hive?

**Comment by cazala** [comment]
why would you do that?

**Comment by vidoza** [comment]
I am not sure what @billygerhard mean but maybe it is because of antivirus app? Coinhive.com domain is on many blacklist and it is automatically blocked.

**Comment by cazala** [comment]
yea but you could simply use xmr-stak and mine on any regular pool (supportxmr, moneroocean, minexmr, etc). Why do you need coinhive if you are not doing web mining? Why would you pay the 30% commission if you can simply mine on any other pool and pay the regular <1% commissions?

**Comment by notgne2** [comment]
it is a very very weird and arguably stupid request, but there's no reason it shouldn't be possible, though it's well out of the scope of this project. If you want something like that it's going to be a totally new codebase and could even end up being a little more complex. If there's a valid reason behind it I'm sure somebody is going to make it, or maybe even already has.

**Comment by billygerhard** [comment]
They take a 30% commission, but they guarantee payout per hash, even if their pool doesn't get the block reward. If you are using coin hive as service on a site, then you probably have servers running like I do, and they are not anywhere near maxed out. If I can add to my coin hive balance with double or triple the hashing power with the same processors, why not. yes, I can just point them to a normal pool, but getting to that payout threshold on a couple servers alone will take lots of time, but if I can get my coin hive balance to the payout amount faster, that would help pay the server bills.

**Comment by cazala** [comment]
fair enough, that's a good answer. this was not the first time someone asked for this, that's why I wanted to know what was the reason behind this request, I was just curious. now answering your question: it's doable, but it's not possible with this proxy. it's basically doing the same thing but the other way around, receiving TCP connections from miners, and connecting to Coinhive via WebSockets, translating stratum message from the miners to Coinhive's custom protocol and vice versa.

**Comment by vidoza** [comment]
I tried to do it with something like that https://www.nginx.com/blog/websocket-nginx/ . But with huge traffic it was very unstable and server stucks after several minutes. Is there anyone who have enought knowlage with websocket, who can create something like that and share with others?

**Comment by cpufanboy** [comment]
You mean kind of like this?
http://xmr.cpufan.club/

It's a 1% commission, fwiw

**Comment by hitmanex** [comment]
@cpufanboy no, not like your noobish site that anyone can make. 

hes asking for mining to coinhive. stop spamming your useless site

**Comment by EZForever** [comment]
@billygerhard  Hi there,
I wonder if you're still interested in this topic, but if you are, I have a good news for you.
I've analyzed almost all JS code from CoinHive last month and made a crypto library and a miner based on that.
See [libcryptohive](https://github.com/EZForever/libcryptohive) and [HiveMiner](https://github.com/EZForever/HiveMiner) for more information.
I hope that these help to you.

P.S. Sorry about my poor English, I'm from China...

**Comment by billygerhard** [comment]
I haven't looked into it in a while, but I will check out your site when I get a chance!
